### ISSUE TITLE
Crash when starting download of update

### ISSUE THREAD

**Issue by mopfattn** [microsoft/appcenter-sdk-android#1282]
### **Description**

Sometimes we get a crash when the AppCenter SDK tries to download an app update. 

```
     Caused by: java.lang.NullPointerException: println needs a message
        at android.util.Log.println_native(Native Method)
        at android.util.Log.e(Log.java:249)
        at com.microsoft.appcenter.utils.AppCenterLog.error(AppCenterLog.java:169)
        at com.microsoft.appcenter.distribute.ReleaseDownloadListener.onError(ReleaseDownloadListener.java:116)
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerReleaseDownloader.onDownloadError(DownloadManagerReleaseDownloader.java:205)
``` 

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Freshly installed an old release of the app
2. Update dialog is shown, option to download the app update is selected
3. Download should start, but the app crashes

### **Details**

1. Which SDK version are you using?
    - 2.4.0
2. Which OS version did you experience the issue on?
    - Android 10
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Google Pixel 1 physical device

```
2019-10-17 17:26:35.850 9434-9434/? D/AppCenterCrashes: The memory running level (20) was saved.
2019-10-17 17:26:37.291 9434-9434/? I/AppCenterDistribute: Launcher activity restarted.
2019-10-17 17:26:37.322 9434-9434/? D/AppCenterUtil: Initialize AppCenter with secret ****8160ad6c
2019-10-17 17:26:37.325 9434-9434/? W/AppCenter: App Center may only be configured once.
2019-10-17 17:26:37.326 9434-9485/? I/AppCenterCrashes: Crashes service has already been enabled.
2019-10-17 17:26:37.360 9434-9434/? D/AppCenterDistribute: Get latest release details...
2019-10-17 17:26:37.360 9434-9434/? D/AppCenterDistribute: Check if we need to report release installation..
2019-10-17 17:26:37.361 9434-9434/? D/AppCenterDistribute: Current release was already reported, skip reporting.
2019-10-17 17:26:37.361 9434-9498/? V/AppCenterDistribute: Calling https://api.appcenter.ms/v0.1/sdk/apps/****************************8160ad6c/releases/latest?release_hash=***...
2019-10-17 17:26:37.361 9434-9498/? V/AppCenterDistribute: Headers: {x-api-token=********************************195e9ca3}
2019-10-17 17:26:38.626 9434-9498/? V/AppCenter: HTTP response status=200 payload={"app_name": ... }
2019-10-17 17:26:38.628 9434-9434/? D/AppCenterDistribute: Check if latest release is more recent.
2019-10-17 17:26:38.628 9434-9434/? D/AppCenterDistribute: Latest release more recent=true
2019-10-17 17:26:38.629 9434-9434/? D/AppCenterDistribute: Latest release is more recent.
2019-10-17 17:26:38.629 9434-9434/? D/AppCenterDistribute: Show default update dialog.
2019-10-17 17:26:38.881 9434-9434/? D/AppCenterDistribute: Show default update dialog.
2019-10-17 17:26:38.891 9434-9434/? V/AppCenterDistribute: Already checking or checked latest release.
2019-10-17 17:27:01.218 9434-9434/? D/AppCenterDistribute: Schedule download...
2019-10-17 17:27:01.220 9434-9942/? D/AppCenterDistribute: Start downloading new release from https://rink.hockeyapp.net/api/2/apps/***/app_versions/3?format=apk&avtoken=***&download_origin=hockeyapp&mctoken=***
2019-10-17 17:27:01.299 9434-9942/? D/AppCenterDistribute: Start download 2.0 (200102) update.
2019-10-17 17:27:02.618 9434-9942/? D/AppCenterDistribute: Check download id=437
2019-10-17 17:27:02.633 9434-9930/? E/AppCenterDistribute: Failed to download update id=437
    java.lang.IllegalStateException
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerUpdateTask.doInBackground(DownloadManagerUpdateTask.java:52)
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerUpdateTask.doInBackground(DownloadManagerUpdateTask.java:20)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
    
    
    --------- beginning of crash
2019-10-17 17:27:02.635 9434-9930/? E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #5
    Process: ***, PID: 9434
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$4.done(AsyncTask.java:399)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.NullPointerException: println needs a message
        at android.util.Log.println_native(Native Method)
        at android.util.Log.e(Log.java:249)
        at com.microsoft.appcenter.utils.AppCenterLog.error(AppCenterLog.java:169)
        at com.microsoft.appcenter.distribute.ReleaseDownloadListener.onError(ReleaseDownloadListener.java:116)
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerReleaseDownloader.onDownloadError(DownloadManagerReleaseDownloader.java:205)
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerUpdateTask.doInBackground(DownloadManagerUpdateTask.java:65)
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerUpdateTask.doInBackground(DownloadManagerUpdateTask.java:20)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
2019-10-17 17:27:02.730 9434-9930/? D/AppCenterCrashes: Saving uncaught exception.
2019-10-17 17:27:02.738 9434-9930/? D/AppCenterCrashes: Saved JSON content for ingestion into /data/user/0/***/files/error/079f03d8-5155-45c8-8d50-3e6eecef6f2d.json
2019-10-17 17:27:02.739 9434-9930/? D/AppCenterCrashes: Saved stack trace as is for client side inspection in /data/user/0/***/files/error/079f03d8-5155-45c8-8d50-3e6eecef6f2d.throwable stack trace:java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$4.done(AsyncTask.java:399)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.NullPointerException: println needs a message
        at android.util.Log.println_native(Native Method)
        at android.util.Log.e(Log.java:249)
        at com.microsoft.appcenter.utils.AppCenterLog.error(AppCenterLog.java:169)
        at com.microsoft.appcenter.distribute.ReleaseDownloadListener.onError(ReleaseDownloadListener.java:116)
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerReleaseDownloader.onDownloadError(DownloadManagerReleaseDownloader.java:205)
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerUpdateTask.doInBackground(DownloadManagerUpdateTask.java:65)
        at com.microsoft.appcenter.distribute.download.manager.DownloadManagerUpdateTask.doInBackground(DownloadManagerUpdateTask.java:20)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
2019-10-17 17:27:02.739 9434-9485/? D/AppCenter: Cleared pending log states
2019-10-17 17:27:02.740 9434-9485/? D/AppCenter: Channel completed shutdown.
```

**Comment by russelarms** [comment]
Hi @mopfattn! Thanks for reaching out.
I'll have a look and get back to you soon.

**Comment by MatkovIvan** [comment]
Hi @mopfattn,

Thanks for the report. The fix is already merged and we're going to release SDK with the patch soon.
The problem was in error logging when downloading an update fails.

**Comment by guperrot** [comment]
This bug has been fixed in version 2.4.1 but we recommend skipping to version 2.5.0 as we fixed critical issues affecting all 2.4.x versions.
